### PR TITLE
RavenDB-20465 Move Studio Configuration on react

### PR DESCRIPTION
**Pull request by kalczur**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-20465/Move-Studio-Configuration-on-react

### Additional description

Moved both database and global studio config.
Fixed form number input. (allow set value to null).
Now FormSwitch has default color="primary"

### Type of change

- New feature

### How risky is the change?

- Low

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- It has been verified by manual testing

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

image
image

### CODE REVIEW

File: src/Raven.Studio/typescript/components/common/studioConfiguration/StudioConfigurationUtils.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { SelectOption } from "../Select";
+
+export const allStudioEnvironments = exhaustiveStringTuple<StudioEnvironment>()(
+    "None",
+    "Development",
+    "Testing",
+    "Production"
+);
+
+export const environmentOptions: SelectOption<StudioEnvironment>[] = allStudioEnvironments.map((environment) => ({
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/studioConfiguration/StudioConfigurationUtils.ts]: rename to studioEnvironmentOptions ?

**Review by ml054 — APPROVED** [review]

**Comment by ppekrol** [comment]
Please rebase

File: src/Raven.Studio/typescript/components/common/studioConfiguration/StudioConfigurationUtils.ts (hunk 1 above)
Comment (kalczur) [reply src/Raven.Studio/typescript/components/common/studioConfiguration/StudioConfigurationUtils.ts]: done

**Comment by kalczur** [comment]
> Please rebase

done

**Pull request merged**